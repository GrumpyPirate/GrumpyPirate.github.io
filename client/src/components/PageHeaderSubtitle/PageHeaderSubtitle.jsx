import React from 'react';
import PropTypes from 'prop-types';

import Container from 'components/Layout/Container/Container';
import Row from 'components/Layout/Row/Row';
import Column from 'components/Layout/Column/Column';

import classes from './PageHeaderSubtitle.scss';

const PageHeaderSubtitle = ({ children }) => (
  <h2 className={classes['page-header-subtitle']}>
    <Container>
      <Row justify="center">
        <Column spanSM={10} spanMD={8}>
          {children}
        </Column>
      </Row>
    </Container>
  </h2>
);

PageHeaderSubtitle.propTypes = {
  children: PropTypes.node.isRequired,
};

export default PageHeaderSubtitle;
