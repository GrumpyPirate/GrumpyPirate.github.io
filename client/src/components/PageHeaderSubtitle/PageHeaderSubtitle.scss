@import 'core';

.page-header-subtitle {
  @include heading(4);
  @include make-link-colour(palette(theme-accent-1), $lighten: true);
  text-transform: uppercase;
  margin: 0;
  padding-bottom: 1rem;
  color: mix(palette(theme-light), palette(theme-dark), 40%);

  @include respond-up(md) {
    padding-bottom: 2rem;
  }
}
