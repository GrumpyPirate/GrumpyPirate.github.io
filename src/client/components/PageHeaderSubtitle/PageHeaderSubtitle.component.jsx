import React from 'react';
import PropTypes from 'prop-types';

import Container from 'components/Layout/Container/Container.component';
import Row from 'components/Layout/Row/Row.component';
import Column from 'components/Layout/Column/Column.component';

import classes from './PageHeaderSubtitle.component.scss';

const PageHeaderSubtitle = ({ children }) => (
  <h2 className={classes['page-header-subtitle']}>
    <Container>
      <Row justify="center">
        <Column spanSM={10} spanMD={8}>
          {children}
        </Column>
      </Row>
    </Container>
  </h2>
);

PageHeaderSubtitle.propTypes = {
  children: PropTypes.node.isRequired,
};

export default PageHeaderSubtitle;
