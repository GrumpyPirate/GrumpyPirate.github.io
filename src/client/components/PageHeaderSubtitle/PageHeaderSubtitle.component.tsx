import React, { FunctionComponent } from 'react';
import PropTypes from 'prop-types';

import { ChildrenProps } from 'types';

import Container from 'components/Layout/Container/Container.component';
import Row from 'components/Layout/Row/Row.component';
import Column from 'components/Layout/Column/Column.component';

import classes from './PageHeaderSubtitle.component.scss';

const PageHeaderSubtitle: FunctionComponent<ChildrenProps> = ({ children }: ChildrenProps) => (
  <h2 className={classes['page-header-subtitle']}>
    <Container>
      <Row justify="center">
        <Column spanSM={10} spanMD={8}>
          {children}
        </Column>
      </Row>
    </Container>
  </h2>
);

PageHeaderSubtitle.propTypes = {
  children: PropTypes.node.isRequired,
};

export default PageHeaderSubtitle;
