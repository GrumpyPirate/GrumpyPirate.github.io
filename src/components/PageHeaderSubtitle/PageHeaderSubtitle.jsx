import React from 'react'
import PropTypes from 'prop-types'

const PageHeaderSubtitle = ({ children }) => {
  return (
    <h2 className="h4 mb-0 pb-1 pb-md-2">
      <div className="container-fluid">
        <div className="row justify-content-center">
          <div className="col-12 col-sm-10 col-md-8">
            {children}
          </div>
        </div>
      </div>
    </h2>
  )
}

PageHeaderSubtitle.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ])
}

export default PageHeaderSubtitle
