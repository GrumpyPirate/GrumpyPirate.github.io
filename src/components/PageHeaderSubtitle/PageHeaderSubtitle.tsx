import Color from 'color';
import React, { FunctionComponent } from 'react';
import styled from 'styled-components';

import { Column, Container, Row } from 'components/Grid';
import { createHeading, createLinkStyle, mediaQueries, palette } from 'styles';
import { PropsWithClassName } from 'types/common';

const PageHeaderSubtitle: FunctionComponent<PropsWithClassName> = function ({
  children,
  className,
}) {
  return (
    <h2 className={className}>
      <Container>
        <Row xAlign="center">
          <Column sm={10} md={8}>
            {children}
          </Column>
        </Row>
      </Container>
    </h2>
  );
};

export default styled(PageHeaderSubtitle)`
  ${createHeading(4)}

  color: ${Color(palette.themeLightShades).alpha(0.5).string()};
  margin: 0;
  padding-bottom: 1rem;
  text-transform: uppercase;

  @media ${mediaQueries.md} {
    padding-bottom: 2rem;
  }

  a {
    ${createLinkStyle({
      color: palette.themeAccentLight,
      lighten: true,
    })}
  }
`;
