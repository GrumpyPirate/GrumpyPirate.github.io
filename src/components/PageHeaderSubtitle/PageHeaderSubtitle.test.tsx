import { render } from '@testing-library/react';
import React from 'react';

import PageHeaderSubtitle from './PageHeaderSubtitle';
import fixture from './PageHeaderSubtitle.fixture';

describe('Components', () => {
  describe('PageHeaderSubtitle', () => {
    const props = {
      ...fixture,
    };

    it('should render as expected, without crashing', () => {
      const { container } = render(<PageHeaderSubtitle {...props} />);

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
