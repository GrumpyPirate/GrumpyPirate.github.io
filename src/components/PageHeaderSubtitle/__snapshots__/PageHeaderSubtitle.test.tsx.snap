// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components PageHeaderSubtitle should render as expected, without crashing 1`] = `
.c3 {
  position: relative;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  max-width: 100%;
  min-height: 1px;
  padding-right: 0.75rem;
  padding-left: 0.75rem;
}

.c1 {
  width: 100%;
  padding-right: 0.75rem;
  padding-left: 0.75rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  margin-right: -0.75rem;
  margin-left: -0.75rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 {
  font-family: 'Barlow',sans-serif;
  font-weight: 300;
  -webkit-letter-spacing: 0.2em;
  -moz-letter-spacing: 0.2em;
  -ms-letter-spacing: 0.2em;
  letter-spacing: 0.2em;
  line-height: 1.2;
  margin: 0 0 1rem;
  text-transform: uppercase;
  font-size: 1rem;
  color: rgba(242,241,243,0.5);
  margin: 0;
  padding-bottom: 1rem;
  text-transform: uppercase;
}

.c0 a {
  color: #dc8381;
}

.c0 a:hover,
.c0 a:focus {
  color: hsl(1.3000000000000114,56.5%,75.3%);
}

.c0 a:active {
  color: hsl(1.3000000000000114,56.5%,61.6%);
}

@media (min-width:48em) {
  .c3 {
    padding-right: 0.9375rem;
    padding-left: 0.9375rem;
  }
}

@media (min-width:30em) {
  .c3 {
    -webkit-flex-basis: 83.33333333333334%;
    -ms-flex-preferred-size: 83.33333333333334%;
    flex-basis: 83.33333333333334%;
    max-width: 83.33333333333334%;
  }
}

@media (min-width:48em) {
  .c3 {
    -webkit-flex-basis: 66.66666666666666%;
    -ms-flex-preferred-size: 66.66666666666666%;
    flex-basis: 66.66666666666666%;
    max-width: 66.66666666666666%;
  }
}

@media (min-width:48em) {
  .c1 {
    padding-right: 0.9375rem;
    padding-left: 0.9375rem;
  }
}

@media (min-width:48em) {
  .c2 {
    margin-right: -0.9375rem;
    margin-left: -0.9375rem;
  }
}

@media (min-width:48em) {
  .c0 {
    font-size: 1.5rem;
  }
}

@media (min-width:48em) {
  .c0 {
    padding-bottom: 2rem;
  }
}

<h2
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <div>
          Test
        </div>
      </div>
    </div>
  </div>
</h2>
`;
